FlorianGrollich/Banana
# Test/update auth tests

### Summary

This PR introduces new tests and backend improvements to the login and registration processes.

### Key Changes

- Enhanced `RegistrationForm` tests with fetch mocking and error handling scenarios.
- Refined `LoginForm` with improved error handling for invalid credentials and unexpected errors.
- Implemented token generation and verification using JWT for enhanced security in user authentication.
- Added password confirmation and token generation logic to the user registration flow.

### Testing

- Ensure there are no build errors in the server and it's running correctly on localhost:3200.
- Verify database connections and route handlers are functioning correctly.
- Verify the functionality of the token-based authentication system.

### Review Requests

- Please check that the tests run correctly.
- Please check that there are no build and type errors.
- Please check the dependency setup and the TypeScript configurations.
- Ensure that the .env file is properly set up according to the new dotenv integration.

Thank you!
